src/lucky/tags/custom_tags.cr

```diff
@@ -43,4 +43,10 @@ module Lucky::CustomTags
     tag_attrs = build_tag_attrs(merged_options)
     view << "<#{name}" << tag_attrs << ">"
   end
+
+  private def tag_content_check(content)
+    return if content.nil?
+    return content if typeof(content) == IO::Memory
+    raise Lucky::Exceptions::InvalidTagError.new(content.to_s)
+  end
```

Comment:
What if we do this at compile time instead!?

```
tag_content_check!(content : String)
  {% content.raise “the error message” %}
end

tag_content_check!(content)
  content
end
```

Reply:
Was on mobile. These should be methods :P

Reply:
I like the idea of compile time. I don't think there's a way to really test it in crystal just yet. The only thing though using the method overload way is maybe catching for more than just string? like `div { 1 }` or `div { :taco }`. Does that matter? If not, then I'll start with string. I guess we can always add to it later too.

Reply:
I think you'd need to have some code in spec/support that has an error and then you'd have to shell out to Crystal, run it, and check that it errors and writes to STDERR as expected...but I dunno :P Haven't tried it.

You've got a point about catching stuff though. Maybe we flip it around so `content : IO?` comes first and returns the content, and `content` comes after and raises. Could you try that out on of your apps and see if it still compiles ok? If so that might be perfect because it will stop just about anything. Even stuff like this `span { user }` where someone accidentally passes the object instead of an attribute of the object.

If that doesn't work though then I think catching `String | Int32` is fine for the first pass 👍

Reply:
However, we should explicitly allow `:taco`. That should print a taco emoji :P